### ISSUE TITLE
NoSuchMethodError on JDK6/7

### ISSUE THREAD

**Issue by fmjaeschke** [tomakehurst/wiremock#322]
Using WireMock.reset() method call on JDK6/7 throws folliowng error:
```java
2015-11-12 12:56:27,312 ERROR [WireMock] - Error for /__admin/reset
java.lang.NoSuchMethodError: java.util.concurrent.ConcurrentHashMap.keySet()Ljava/util/concurrent/ConcurrentHashMap$KeySetView;
	at com.github.tomakehurst.wiremock.global.ThreadSafeRequestDelayControl.cancelAllDelays(ThreadSafeRequestDelayControl.java:51)
	at com.github.tomakehurst.wiremock.global.ThreadSafeRequestDelayControl.clearDelay(ThreadSafeRequestDelayControl.java:38)
	at com.github.tomakehurst.wiremock.core.WireMockApp.resetMappings(WireMockApp.java:141)
	at com.github.tomakehurst.wiremock.admin.ResetTask.execute(ResetTask.java:26)
	at com.github.tomakehurst.wiremock.http.AdminRequestHandler.handleRequest(AdminRequestHandler.java:55)
	at com.github.tomakehurst.wiremock.http.AbstractRequestHandler.handle(AbstractRequestHandler.java:38)
	at com.github.tomakehurst.wiremock.jetty6.Jetty6HandlerDispatchingServlet.service(Jetty6HandlerDispatchingServlet.java:98)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at org.mortbay.jetty.servlet.ServletHolder.handle(ServletHolder.java:511)
	at org.mortbay.jetty.servlet.ServletHandler.handle(ServletHandler.java:401)
	at org.mortbay.jetty.handler.ContextHandler.handle(ContextHandler.java:766)
	at org.mortbay.jetty.handler.HandlerCollection.handle(HandlerCollection.java:114)
	at org.mortbay.jetty.handler.HandlerWrapper.handle(HandlerWrapper.java:152)
	at org.mortbay.jetty.Server.handle(Server.java:326)
	at org.mortbay.jetty.HttpConnection.handleRequest(HttpConnection.java:542)
	at org.mortbay.jetty.HttpConnection$RequestHandler.headerComplete(HttpConnection.java:928)
	at org.mortbay.jetty.HttpParser.parseNext(HttpParser.java:549)
	at org.mortbay.jetty.HttpParser.parseAvailable(HttpParser.java:212)
	at org.mortbay.jetty.HttpConnection.handle(HttpConnection.java:404)
	at org.mortbay.jetty.bio.SocketConnector$Connection.run(SocketConnector.java:228)
	at com.github.tomakehurst.wiremock.jetty6.DelayableSocketConnector$1.run(DelayableSocketConnector.java:49)
	at org.mortbay.thread.QueuedThreadPool$PoolThread.run(QueuedThreadPool.java:582)

com.github.tomakehurst.wiremock.client.VerificationException: Expected status 200 for http://localhost:59310/__admin/reset but was 500

	at com.github.tomakehurst.wiremock.client.HttpAdminClient.postJsonAssertOkAndReturnBody(HttpAdminClient.java:156)
	at com.github.tomakehurst.wiremock.client.HttpAdminClient.resetMappings(HttpAdminClient.java:86)
	at com.github.tomakehurst.wiremock.client.WireMock.resetMappings(WireMock.java:101)
	at com.github.tomakehurst.wiremock.client.WireMock.reset(WireMock.java:105)
```
If ThreadSafeRequestDelayControl.java Line 26 is changed from
```java
private ConcurrentHashMap<Thread, Object> threadsBeingDelayed = new ConcurrentHashMap<Thread, Object>();
```
to
```java
private Map<Thread, Object> threadsBeingDelayed = new ConcurrentHashMap<Thread, Object>();
```
the error should be gone.

See also https://gist.github.com/AlainODea/1375759b8720a3f9f094

**Comment by Mahoney** [comment]
Duplicate of #309

**Comment by tomakehurst** [comment]
I've just released v1.58, which just fixes this problem.
